Tensile/Source/client/source/HardwareMonitor.cpp

```diff
@@ -386,9 +390,178 @@ namespace Tensile
                 else
                     m_fanValues[i] += newValue;
             }
+            
+            rsmi_gpu_metrics_t gpuMetrics;
+            auto status = rsmi_dev_gpu_metrics_info_get(m_smiDeviceIndex, &gpuMetrics);
+            if(status != RSMI_STATUS_SUCCESS)
+            {
+                m_freqValues[0]        = std::numeric_limits<uint16_t>::max();
```

Comment:
The size of m_freqValues can be 0 here, right?
Is that safe to assign value to m_freqValues[0] here?

Reply:
Good catch, If rsmi_dev_gpu_metrics_info_get returns false in the first attempt, the m_freqValues is of size 0. We can check the size of arrays first.

Reply:
This ties in with my question earlier, where for the other metrics we pre-size the vector so we wouldn't run into the case where the array is empty. Need to either fix it to be consistent with the other metrics or handle the edge case here.

Reply:
Good catch. it might be an issue, changing to something like this.
...
m_freqValues.resize(1);
m_powerValues.resize(1);
m_temperatureValues.resize(1);
..
void HardwareMonitor::collectOnce() 
{
......
if(status != RSMI_STATUS_SUCCESS)
{
                m_freqValues.at(0)        = std::numeric_limits<uint16_t>::max();
                m_powerValues.at(0)       = std::numeric_limits<uint16_t>::max();
                m_temperatureValues.at(0) = std::numeric_limits<uint16_t>::max();
}
else
{
                // size of m_freqValues,m_powerValues,m_temperatureValues is same.   
                if(m_freqValues[0] != std::numeric_limits<uint16_t>::max())  
                {    
                    m_freqValues.push_back(gpuMetrics.average_gfxclk_frequency);
                    m_powerValues.push_back(gpuMetrics.average_socket_power);
                    m_temperatureValues.push_back(gpuMetrics.temperature_hotspot);
                }
}
...